manifests/init.pp

```diff
@@ -8,12 +8,16 @@
 #
 # @param mutt_ensure String to pass to the `mutt` package ensure attribute
 #
+# @param inet_protocols
+#   The protocols to use when enabling the service
+#
 # @author Trevor Vaughan <tvaughan@onyxpoint.com>
 #
 class postfix (
-  Boolean $enable_server = false,
-  String $postfix_ensure = simplib::lookup('simp_options::package_ensure', { 'default_value' => 'installed' }),
-  String $mutt_ensure = simplib::lookup('simp_options::package_ensure', { 'default_value' => 'installed' }),
+  Boolean                $enable_server  = false,
+  Simplib::PackageEnsure $postfix_ensure = simplib::lookup('simp_options::package_ensure', { 'default_value' => 'installed' }),
```

Comment:
Individual packages should not use the `Simplib::PackageEnsure`, because you could also set a version string.

Reply:
To @jeefberkey's point: Is there any reason we can't just update the definition of `Simplib::PackageEnsure` to be something more inclusive, like:

```puppet
type Simplib::PackageEnsure = Variant[
  Enum['absent', 'held',...],
  # chainsawed from yum::versionlockstring (http://rubular.com/r/gujnH0CGrq)
  Pattern[^[0-9]+(\.[0-9])*(-(([0-9a-zA-Z_\*-.]+)))?],
]
```

Reply:
Summarizing offline conversations with @jeefberkey and later @trevor-vaughan:
* `Simplib::PackageEnsure` is used elsewhere to govern multiple packages, so would be inappropriate to modify it to permit single-version strings.  If needed, a new type alias would have to be created for this purpose.
* The long-standing SIMP convention has been to avoid pinning individual package versions, except in exceptional circumstances (e.g., volatile software like elasticsearch or GitLab).  Other considerations that have informed this opinion include the many unintended consequences of mixing pinned versions with moving updates, and the maintenance nightmare (for both developers and users) that exposing the versions of many-to-all packages would entail.